### DISCUSSION TITLE
Why does rx.call_script require a return statement.

### DISCUSSION THREAD

**Discussion by abdullahkhalids** [reflex-dev/reflex#2161] [Q&A] [answered]
In the [docs](https://reflex.dev/docs/api-reference/browser_javascript/), there is the following example of using `rx.call_script` inside a State function.

```python
class SoundEffectState(rx.State):
    @rx.background
    async def delayed_play(self):
        await asyncio.sleep(1)
        return rx.call_script("playFromStart(button_sfx)")
```
I have played around with it, and it seems that `call_script` is not executed if it is not part of a return statement. **Why?**

My particular use case is that I am adding a html canvas via `rx.html` and controlling it using browser javascript, called from event handlers of reflex buttons and fields. Inside one those event handlers, I would ideally like to go, "python code -> call_script -> more python code". But it seems this is not possible.

**Comment by masenf** [comment, marked as answer]
To send the event to the client and continue running on the backend, use `yield` instead of `return`.

Since the event has to run in the browser the event handler needs to return or yield the call_script from the backend.

If you need the call_script to execute on the frontend immediately, then the event handler must be decorated with `@rx.background`, otherwise the frontend will not process the yielded events until the current event handler returns its final update.

**Reply by abdullahkhalids** [reply]
Thanks. I kind of get the logic. Will test out to see if I can get a basic example working.
